### PR TITLE
[REVIEW] Fix single column `MultiIndex` issue in `sort_index`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #10954 

This PR fixes the conversion of a single column `MultIndex` into `Index` in `sort_index`.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -2498,7 +2498,11 @@ def set_index(
 
         if len(columns_to_add) == 0:
             raise ValueError("No valid columns to be added to index.")
-        elif len(columns_to_add) == 1:
+        elif (
+            len(columns_to_add) == 1
+            and len(keys) == 1
+            and not isinstance(keys[0], (cudf.MultiIndex, pd.MultiIndex))
```
Comment (vyasr) [inline python/cudf/cudf/core/dataframe.py:2504]: In the interest of getting this fixed quickly in this version, I'm fine with pushing this through, but we can definitely make this more efficient. If a user provides a MultiIndex as input, for instance, I think we can just set it directly (after error checking) without allocating a new object. Indexes are immutable. Then converting from pandas objects up front should let us use more internals of MultiIndex to do these kinds of checks I would think.

**Review by vyasr — APPROVED** [review]
LGTM, thanks! I see room for improvement in the function, but nothing related to what you're fixing or pressing enough to delay this PR.

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py:2504]: Right, I'll make this kind of a fix for `22.08`. I'll go ahead and merge this as-is for `22.06`.

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**